explore/src/main/scala/explore/config/ITCRequestsQueue.scala

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) 2016-2021 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package explore.itc
+
+import cats._
+import cats.effect._
+import cats.effect.std.Queue
+import cats.effect.syntax.all._
+import cats.syntax.all._
+import clue.TransactionalClient
+import clue.data.Input
+import clue.data.syntax._
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.collection._
+import eu.timepit.refined.types.numeric.PosBigDecimal
+import explore.common.ITCQueriesGQL._
+import explore.model.AirMassRange
+import explore.model.ConstraintSet
+import explore.modes.GmosNorthSpectroscopyRow
+import explore.modes.SpectroscopyModeRow
+import explore.schemas.ITC
+import explore.schemas.itcschema.implicits._
+import fs2.Stream
+import japgolly.scalajs.react.Reusability
+import japgolly.scalajs.react._
+import lucuma.core.enum.StellarLibrarySpectrum
+import lucuma.core.enum._
+import lucuma.core.math.MagnitudeValue
+import lucuma.core.math.Wavelength
+import lucuma.core.model.Magnitude
+import lucuma.core.model.SpatialProfile
+import lucuma.core.model.SpectralDistribution
+import org.typelevel.log4cats.Logger
+
+import scala.concurrent.duration._
+
+final case class ITCRequest[F[_]](
+  mode:          InstrumentModes,
+  wavelength:    Wavelength,
+  signalToNoise: PosBigDecimal,
+  callback:      SpectroscopyITCQuery.Data => F[Unit]
+)
+
+final case class ITCRequestsQueue[F[_]](f: Queue[F, Option[ITCRequest[F]]]) {
+  def run(implicit
+    E: Temporal[F],
+    L: Logger[F],
+    T: TransactionalClient[F, ITC]
+  ): F[Unit] =
+    Stream
+      .fromQueueNoneTerminated(f)
+      .meteredStartImmediately(500.milliseconds) // Limit how fasts requests are created
+      .evalMap(f => ITCRequestsQueue.doRequest(f))
+      .compile
+      .drain
```

Comment:
I think that limiting concurrency might be a more fitting solution than using a Stream. We could send N requests simultaneously initially, and make sure a new one isn't started until another is done.

Looking around on how to achieve this I found this which uses a semaphore: https://gist.github.com/gvolpe/44e2263f9068efe298a1f30390de6d22.

But we can try that in another PR.